domain/upgrade/types.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package upgrade
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/utils/v3"
+)
+
+// UUID represents a upgrade unique identifier.
+type UUID string
+
+// Validate ensures the consistency of the UUID.
+func (u UUID) Validate() error {
+	if u == "" {
+		return errors.New("empty uuid")
+	}
+	if !utils.IsValidUUIDString(string(u)) {
+		return errors.Errorf("invalid uuid %q", u)
```

Comment:
Perhaps `errors.NotValidf`?

Reply:
I don't think it matters in this case, but might be good for consistency.